Sage/carbon
# CARBON-898 Demo site number props

# Description
* Add check to code builder - if Number prop value is empty string, don't add prop code. This prevents code from breaking with invalid JSX error.

* Address linting errors in component action file
* Move `value` to inputDefinition `hiddenProps` for all inputs
* Use Number components for all Number props in demos

# TODO

# Screenshots / Gifs

![numbers](https://user-images.githubusercontent.com/9549240/27855467-8be1b788-6162-11e7-8242-933a3bf08344.gif)

# Testing Instructions
* Check Carousel and Button toggle functionality in Demo Site.
* Check All input components (e.g. Textbox, Dropdown, Number,etc.) to see that `value` still correctly displayed after user input.
* Check components with Number props (esp Carousel slideIndex, Pages slideIndex, Flash timeout, Decimal precision).
